sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventData.java

```diff
@@ -279,99 +321,432 @@ public EventData addContext(String key, Object value) {
     }
 
     /**
-     * A collection of properties populated by Azure Event Hubs service.
+     * Gets the content type of the message.
+     *
+     * <p>
+     * Optionally describes the payload of the message, with a descriptor following the format of RFC2045, Section 5,
+     * for example "application/json".
+     * </p>
+     * @return The content type of the {@link EventData}.
+     */
+    public String getContentType() {
+        return amqpAnnotatedMessage.getProperties().getContentType();
+    }
+
+    /**
+     * Sets the content type of the {@link EventData}.
+     *
+     * <p>
+     * Optionally describes the payload of the message, with a descriptor following the format of RFC2045, Section 5,
+     * for example "application/json".
+     * </p>
+     *
+     * @param contentType RFC2045 Content-Type descriptor of the message.
+     *
+     * @return The updated {@link EventData}.
+     */
+    public EventData setContentType(String contentType) {
+        amqpAnnotatedMessage.getProperties().setContentType(contentType);
+        return this;
+    }
+
+    /**
+     * Gets a correlation identifier.
+     * <p>
+     * Allows an application to specify a context for the message for the purposes of correlation, for example
+     * reflecting the MessageId of a message that is being replied to.
+     * </p>
+     *
+     * @return The correlation id of this message.
+     */
+    public String getCorrelationId() {
```

Comment:
Not sure if EH needs this.

Reply:
yes; `CorrelationId` is one of the three new top-level properties that we're exposing.